src/test/kotlin/ecommerce/controller/ProductControllerTest.kt

```diff
@@ -0,0 +1,364 @@
+    fun create() {
+        val product = Product(name = "product1", price = 1.5, imageUrl = "https://www.product.com/image/1")
+        jdbcProductDao.insert(product)
+    }
+
+    @BeforeEach
+    fun setUp() {
+        jdbcProductDao = JdbcProductDAO(jdbcTemplate)
+        productService = ProductService(jdbcProductDao)
+
+        jdbcTemplate.execute("DROP TABLE product CASCADE")
+        jdbcTemplate.execute(
+            """CREATE TABLE product (
+                         id          LONG    NOT NULL AUTO_INCREMENT,
+                         name        varchar(255)    NOT NULL,
+                         price       DOUBLE  NOT NULL,
+                         imageUrl    TEXT    NOT NULL,
+                         PRIMARY KEY (id)
+                    );""",
+        )
+
+        val query = """INSERT INTO product (name, price, imageUrl) VALUES ('Iron Man', 1000, 'https://alexnsan.comics/imageurl/1');
+                    INSERT INTO product (name, price, imageUrl) VALUES ('X-men', 1000, 'https://alexnsan.comics/imageurl/2');
+                    INSERT INTO product (name, price, imageUrl) VALUES ('Superman', 1000, 'https://alexnsan.comics/imageurl/3');
+                    INSERT INTO product (name, price, imageUrl) VALUES ('Naruto', 1000, 'https://alexnsan.comics/imageurl/4');
+                    INSERT INTO product (name, price, imageUrl) VALUES ('Full Metal Alchemist', 1000, 'https://alexnsan.comics/imageurl/5');"""
+        jdbcTemplate.batchUpdate(query)
+
+        controller = ProductController(productService)
+    }
+
+    @Test
+    fun `create() - should insert product and return 201 when form is valid`() {
+        val response =
+            RestAssured
+                .given().log().all()
+                .body(Product(name = "product1 [new]", price = 1.5, imageUrl = "https://www.product.com/image/1"))
+                .contentType(ContentType.JSON)
+                .`when`().post("/api/products")
+                .then().log().all().extract()
+
+        Assertions.assertThat(response.statusCode()).isEqualTo(HttpStatus.CREATED.value())
+    }
+
+    @Test
+    fun `create() - should return 400 when name is blank`() {
+        val response =
+            RestAssured
+                .given().log().all()
+                .body(Product(name = "", price = 1.5, imageUrl = "https://www.product.com/image/1"))
+                .contentType(ContentType.JSON)
+                .`when`().post("/api/products")
+                .then().log().all().extract()
+
+        val targets =
+            listOf(
+                "Contains unallowed character",
+                "Product name is required",
+                "Must be no more than 15 characters, including spaces",
+            )
+        val resBody = response.jsonPath().getMap<String, String>("errors")
+        val actual = resBody["name"]
+        Assertions.assertThat(response.statusCode()).isEqualTo(HttpStatus.BAD_REQUEST.value())
+        Assertions.assertThat(actual).isIn(targets)
```

Comment:
You've made a lot of tests 👍 
In certain test classes, `assertThat` is used right away while in others, it's called with `Assertions` before it.
Why don't we keep it the same?

This might seem minor, but I like to keep things consistent as much as I can, since we never know where things might go wrong when we don't keep the consistency.